MIGRATION_GUIDE.md

```diff
@@ -2,7 +2,42 @@
 
 ## Migrate from `authmosphere@1.0.x` to `authmosphere@2.0.y`
 
-**TODO**
+### Upgrade
+
+* call `npm install --save authmosphere@~2.0.0`
+
+### General changes
+
+* All exported functions got support for a custom Logger. Providing a logger is optional.
+  Any logger need to satisfy the [Logger](./src/types/Logger.ts) interface.
+  * If a logging framework does not satisfy the interface, it need to be wrapped for authmosphere.
+* To keep arguments lists short, `option` objects were introduced to group a number of (mostly) optional parameters.
+
+
+### express middlewares
+
+* `handleOAuthRequestMiddleware` was renamed to [`authenticationMiddleware`](./src/express-tooling.ts)
+  * config parameter `MiddlewareOptions` was renamed to `AuthenticationMiddlewareOptions`
+  * an optional logger can be provided ([`Logger`](./src/types/Logger.ts))
+  * an optional `onNotAuthenticatedHandler` can be provided, it helps to customize handling the case authentication fails
+
+* `requireScopesMiddleware`
+  * added optional `options` object of type [`ScopeMiddlewareOptions`](./src/types/ScopeMiddlewareOptions.ts)
+    * an optional logger can be provided ([`Logger`](./src/types/Logger.ts))
+    * an optional `onAuthorizationFailedHandler` can be provided, it helps to customize handling the case authentication fails
+  * moved `precedenceOptions` parameter into `options` parameter
+    * `precedenceErrorHandler` got removed from [`PrecedenceOptions`](./src/types/Precedence.ts).
+      `onAuthorizationFailedHandler` should be used instead.
```

Comment:
Is "instead" really correct here, isnt it a different use case?

Reply:
Differences will be explained in the documentation